src/code/z_eff_blure.c

```diff
@@ -66,40 +68,977 @@ void func_800A81F0(EffectBlure* this, Vec3f* p1, Vec3f* p2) {
+            if (this->drawMode == 1) {
+                alphaRatio = (f32)this->elements[i].timer / (f32)this->elemDuration;
+                gDPSetPrimColor(POLY_XLU_DISP++, 0x00, 0x80, this->altPrimColor.r, this->altPrimColor.g,
+                                this->altPrimColor.b, this->altPrimColor.a * (1.0f - alphaRatio));
+                gDPPipeSync(POLY_XLU_DISP++);
+            }
+
+            if (1) {} // Necessary to match
+
+            gSPVertex(POLY_XLU_DISP++, &vtx[j], 4, 0);
+            gSP2Triangles(POLY_XLU_DISP++, 0, 1, 3, 0, 0, 3, 2, 0);
+
+            if (this->flags & 4) {
+                sp1B0.x = ((f32)vtx[4 * i + 0].v.ob[0] + (f32)vtx[4 * i + 1].v.ob[0]) * 0.5f;
+                sp1B0.y = ((f32)vtx[4 * i + 0].v.ob[1] + (f32)vtx[4 * i + 1].v.ob[1]) * 0.5f;
+                sp1B0.z = ((f32)vtx[4 * i + 0].v.ob[2] + (f32)vtx[4 * i + 1].v.ob[2]) * 0.5f;
+                sp1A4.x = ((f32)vtx[4 * i + 2].v.ob[0] + (f32)vtx[4 * i + 3].v.ob[0]) * 0.5f;
+                sp1A4.y = ((f32)vtx[4 * i + 2].v.ob[1] + (f32)vtx[4 * i + 3].v.ob[1]) * 0.5f;
+                sp1A4.z = ((f32)vtx[4 * i + 2].v.ob[2] + (f32)vtx[4 * i + 3].v.ob[2]) * 0.5f;
+
+                Math_Vec3f_Diff(&sp1A4, &sp1B0, &sp198);
+                scale = sqrtf(SQ(sp198.x) + SQ(sp198.y) + SQ(sp198.z));
+
+                if (fabsf(scale) > 0.0005f) {
+                    scale = 1.0f / scale;
+                    Math_Vec3f_Scale(&sp198, scale);
+
+                    SkinMatrix_SetTranslate(&sp154, sp1B0.x, sp1B0.y, sp1B0.z);
+                    SkinMatrix_SetRotateAroundVec(&sp114, 0x3FFF, sp198.x, sp198.y, sp198.z);
+                    SkinMatrix_MtxFMtxFMult(&sp154, &sp114, &spD4);
+                    SkinMatrix_SetTranslate(&sp154, -sp1B0.x, -sp1B0.y, -sp1B0.z);
+                    SkinMatrix_MtxFMtxFMult(&spD4, &sp154, &sp94);
+
+                    mtx = SkinMatrix_MtxFToNewMtx(gfxCtx, &sp94);
+                    if (mtx == NULL) {
+                        break;
+                    }
+
+                    gSPMatrix(POLY_XLU_DISP++, mtx, G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+                    gSPVertex(POLY_XLU_DISP++, &vtx[j], 4, 0);
+                    gSP2Triangles(POLY_XLU_DISP++, 0, 1, 3, 0, 0, 3, 2, 0);
+                    gSPMatrix(POLY_XLU_DISP++, &D_801D1DE0, G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+                }
+            }
+
+            j += 4;
+        }
+    }
+
+    CLOSE_DISPS(gfxCtx);
+}
+
+Vtx_t D_801AE26C[] = {
+    VTX_T(0, 0, 0, 0, 1024, 0xFF, 0xFF, 0xFF, 0xFF),
+    VTX_T(0, 0, 0, 0, 0, 0xFF, 0xFF, 0xFF, 0xFF),
+    VTX_T(0, 0, 0, 2048, 1024, 0xFF, 0xFF, 0xFF, 0xFF),
+    VTX_T(0, 0, 0, 2048, 0, 0xFF, 0xFF, 0xFF, 0xFF),
+};
+
+Vtx_t D_801AE2AC[] = {
+    VTX_T(0, 0, 0, 2048, 1024, 0xFF, 0xFF, 0xFF, 0xFF),
+    VTX_T(0, 0, 0, 2048, 0, 0xFF, 0xFF, 0xFF, 0xFF),
+    VTX_T(0, 0, 0, 2048, 1024, 0xFF, 0xFF, 0xFF, 0xFF),
+    VTX_T(0, 0, 0, 2048, 0, 0xFF, 0xFF, 0xFF, 0xFF),
+};
```

Comment:
```suggestion
static Vtx_t D_801AE26C[] = {
    VTX_T(0, 0, 0, 0, 1024, 0xFF, 0xFF, 0xFF, 0xFF),
    VTX_T(0, 0, 0, 0, 0, 0xFF, 0xFF, 0xFF, 0xFF),
    VTX_T(0, 0, 0, 2048, 1024, 0xFF, 0xFF, 0xFF, 0xFF),
    VTX_T(0, 0, 0, 2048, 0, 0xFF, 0xFF, 0xFF, 0xFF),
};

static Vtx_t D_801AE2AC[] = {
    VTX_T(0, 0, 0, 2048, 1024, 0xFF, 0xFF, 0xFF, 0xFF),
    VTX_T(0, 0, 0, 2048, 0, 0xFF, 0xFF, 0xFF, 0xFF),
    VTX_T(0, 0, 0, 2048, 1024, 0xFF, 0xFF, 0xFF, 0xFF),
    VTX_T(0, 0, 0, 2048, 0, 0xFF, 0xFF, 0xFF, 0xFF),
};
```
This is a very small set of Vtx data so it's less of a big deal. But in general, MM has taken a more hardline approach to making sure all possible asset data, no matter how small or trivial, be extracted if possible. I think my preference would be to have this extracted, but I'll leave this up to you given the context

Reply:
Yeah. We are requiring every Vtx and texture to be extracted.
This is code tho, so the XML to extract them will be weird. If you need help, we could try to figure out a way to do it in discord.

Reply:
Imho basic vertices like these should be treated as an exception, especially in this case since they don't correspond to anything on their own and are just used as "base" vertices which are then set through code to contain the data needed, if that makes sense (see `EffectBlure_DrawSimple` for instance).

Reply:
That being said, we can try to extract them if it's really necessary, but we would first need ZAPD to handle `VTX_T` somehow, and any vertices that are in-function static will be difficult to extract to say the least

Reply:
In-function static Vtx shouldn't be a problem tbh, since ZAPD can extract Vtx to standalone `.vtx.inc` files. It would look like something like this:
```c
void function() {
  static Vtx someVtx[] = {
    #include "path/somevtx.vtx.inc"
  }

  // ...
}
```
Currently this isn't 100% reliable because of a few ZAPD bugs which prevent the extraction to standalone files. Those bugs should be fixed in PR https://github.com/zeldaret/ZAPD/pull/151.

Second, what's the difference between a VTX and a VTX_T? The macro looks similar enough. From a extraction pov looks like ZAPD would only need to replace `VTX` to `VTX_T`.

Finally, I can see why vertices like this could be treated as an exception, but I think this needs a further discussion.